blocks/news-feed/news-feed.js

```diff
@@ -0,0 +1,104 @@
+import ffetch from '../../scripts/ffetch.js';
+import { createOptimizedPicture, decorateIcons } from '../../scripts/lib-franklin.js';
+
+function buildPost(post, eager) {
+  const postCard = document.createElement('div');
+  postCard.classList.add('news-card');
+
+  const description = post.content.querySelector('h1 + p');
+  const title = post.content.querySelector('h1').textContent;
+
+  postCard.innerHTML = `
+      <div class="news-image">
+        <a href="${post.path}">${createOptimizedPicture(post.image, `Teaser image for ${title}`, eager).outerHTML}</a>
+      </div>
+      <div class="news-content">
+        <a class="post-title" href="${post.path}">${title}</a>
+        <a class="post-description" href="${post.path}">
+          <p>${description ? description.textContent : post.description}</p>
+          <span>Read More</span>
+        </a>
+      </div>
+    </a>
+  `;
+
+  return postCard;
+}
+
+async function buildNewsFeed(ul, pageNum, pageControl) {
```

Comment:
`pageControl` is an HTML Element. For clarity I'd rename it into `pageContent` (initially I thought it's a controller)